### PR TITLE
Add zero filter in save functions

### PR DESCRIPTION
**Pull request by devsjc**
Fixes Satip saving datasets with high numbers of zeros, breaking the forecaster.

### CODE REVIEW

**Review by jacobbieker — APPROVED** [review]

File: satip/utils.py [hunk 1]
```diff
@@ -410,13 +410,26 @@ def get_dataset_from_scene(filename: str, hrv_scaler, use_rescaler: bool, save_d
...
+
     save_file = os.path.join(save_dir, f"{'15_' if using_backup else ''}hrv_{now_time}.zarr.zip")
     log.debug(f"Saving HRV netcdf in {save_file}", memory=get_memory())
     save_to_zarr_to_s3(hrv_dataset, save_file)
     del hrv_dataset
     gc.collect()
     log.debug("Saved HRV to NetCDF", memory=get_memory())
 
+def data_quality_filter(ds: xr.Dataset, threshold_fraction: float = 0.9) -> bool:
+    for var in ds.data_vars:
```
Comment (peterdudfield) [inline satip/utils.py:431]: worth adding a function description?

**Review by peterdudfield — APPROVED** [review]

**Comment by devsjc** [comment]
Chucked a test in too.

**Comment by peterdudfield** [comment]
Some of the python tests seem to fail, should these be failing?

**Comment by devsjc** [comment]
Jacob said that was to do with the eumetstat api endpoint being down?

**Comment by jacobbieker** [comment]
Yeah, the user notification service says the Data store/data tailor is in maintenance until the 23rd, the error isn't with the code here

**Pull request merged**